tests/jwt_test.py

```diff
@@ -49,28 +65,147 @@ def test_refresh_jwt_issuer(network, args):
+
+    with tempfile.NamedTemporaryFile(prefix="ccf", mode="w+") as jwks_fp:
+        json.dump(keys, jwks_fp)
+        jwks_fp.flush()
+        network.consortium.set_jwt_public_signing_keys(
+            primary, issuer.name, jwks_fp.name
+        )
+
+    service_keys = get_jwt_keys(args, primary)
+    assert service_keys["kid1"]["issuers"][issuer.name] == issuer.name
+
+    new_constraint = "https://example.issuer/very/specific"
+    keys["keys"][0]["issuer"] = new_constraint
+    with tempfile.NamedTemporaryFile(prefix="ccf", mode="w+") as jwks_fp:
+        json.dump(keys, jwks_fp)
+        jwks_fp.flush()
+        network.consortium.set_jwt_public_signing_keys(
+            primary, issuer.name, jwks_fp.name
+        )
+
+    service_keys = get_jwt_keys(args, primary)
+    assert service_keys["kid1"]["issuers"][issuer.name] == new_constraint
+
+    network.consortium.remove_jwt_issuer(primary, issuer.name)
+
+
+@reqs.description("Only able to set keys with issuer constraints matching the url")
+def test_jwt_issuer_domain_match(network, args):
+    """Check domains match. Additional subdomains permitted. For example, https://limited.facebook.com
+    may provide keys with issuer constraint https://facebook.com."""
+
+    primary, _ = network.find_nodes()
+
+    issuer = infra.jwt_issuer.JwtIssuer("https://trusted.issuer.com/something")
+    keys = issuer.create_jwks_for_kids(["kid1"])
+
+    with tempfile.NamedTemporaryFile(prefix="ccf", mode="w+") as metadata_fp:
+        json.dump({"issuer": issuer.name}, metadata_fp)
+        metadata_fp.flush()
+        network.consortium.set_jwt_issuer(primary, metadata_fp.name)
+
+    with tempfile.NamedTemporaryFile(prefix="ccf", mode="w+") as jwks_fp:
+        json.dump(keys, jwks_fp)
+        jwks_fp.flush()
+        network.consortium.set_jwt_public_signing_keys(
+            primary, issuer.name, jwks_fp.name
+        )
+
+    service_keys = get_jwt_keys(args, primary)
+    assert service_keys["kid1"]["issuers"][issuer.name] == issuer.name
+
+    keys["keys"][0]["issuer"] = "https://issuer.com"
+
+    with tempfile.NamedTemporaryFile(prefix="ccf", mode="w+") as jwks_fp:
+        json.dump(keys, jwks_fp)
+        jwks_fp.flush()
+        network.consortium.set_jwt_public_signing_keys(
+            primary, issuer.name, jwks_fp.name
+        )
+
+    garbage = ["", "garbage", "https://another.com", "https://issuer.com.domain"]
+    for constraint in garbage:
+        with tempfile.NamedTemporaryFile(prefix="ccf", mode="w+") as jwks_fp:
+            keys["keys"][0]["issuer"] = constraint
+            json.dump(keys, jwks_fp)
+            jwks_fp.flush()
+            try:
+                network.consortium.set_jwt_public_signing_keys(
+                    primary, issuer.name, jwks_fp.name
+                )
+            except infra.proposal.ProposalNotAccepted:
+                pass
+            else:
+                assert False, f"Constraint {constraint} must not be allowed"
+
```

Comment:
```suggestion

    network.consortium.remove_jwt_issuer(primary, issuer.name)
```

The other tests remove their temporary issuer to clean up the network before closing - should this do the same?